### PR TITLE
Preserve BootOrder when flashing

### PR DESCRIPTION
**Pull request by crawfxrd**
Save the data for `BootOrder` and `Boot####` before flashing and write it back after flashing.

Resolves: #6

### CODE REVIEW

File: src/app/bios.rs [hunk 1]
```diff
@@ -390,9 +391,32 @@ impl Component for BiosComponent {
...
+                println!("Failed to preserve boot order");
+            }
+
+            let cmd = format!("{} {} bios flash", FIRMWARENSH, FIRMWAREDIR);
             let status = shell(&cmd)?;
+
+            if order.is_ok() {
+                if set_boot_order(&order.unwrap()).is_ok() {
+                    for (num, data) in boot_options {
+                        set_boot_item(num, &data)?;
```
Comment (jackpot51) [inline src/app/bios.rs]: Do we want to fail in this case?

File: src/app/bios.rs [hunk 2]
```diff
@@ -390,9 +391,32 @@ impl Component for BiosComponent {
...
             find(FIRMWARENSH)?;
 
-            let cmd = format!("{} {} bios flash", FIRMWARENSH, FIRMWAREDIR);
+            let mut boot_options: Vec<(u16, Vec<u8>)> = vec!();
 
+            let order = get_boot_order();
+            if order.is_ok() {
+                println!("Preserving boot order");
+                for num in order.clone().unwrap() {
+                    boot_options.push((num, get_boot_item(num)?));
```
Comment (jackpot51) [inline src/app/bios.rs]: Do we want to fail in this case?

**Comment by jackpot51** [comment]
This looks pretty good. I just have two questions - do we want the flashing process to fail in the event set_boot_item or get_boot_item returns an error? We could either do that or print an error

File: src/app/bios.rs (hunk 2 above)
Comment (crawfxrd) [reply src/app/bios.rs]: Probably not here. Should continue as a best-effort to restore the boot order at this point.

File: src/app/bios.rs (hunk 1 above)
Comment (crawfxrd) [reply src/app/bios.rs]: Probably not here. Should continue as a best-effort to restore the boot order at this point.

File: src/app/bios.rs (hunk 2 above)
Comment (crawfxrd) [reply src/app/bios.rs]: Will continue as a best-effort.

**Pull request merged**